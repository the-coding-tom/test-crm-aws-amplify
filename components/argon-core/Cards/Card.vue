<template>
  <div
    :class="[
      {'card-lift--hover': hover},
      {'shadow': shadow},
      {[`shadow-${shadowSize}`]: shadowSize},
      {[`bg-gradient-${gradient}`]: gradient},
      {[`bg-${type}`]: type}
    ]"
    class="card">

    <slot name="image"/>
    <div
      v-if="$slots.header"
      :class="headerClasses"
      class="card-header">
      <slot name="header"/>
    </div>
    <div
      v-if="!noBody"
      :class="bodyClasses"
      class="card-body">
      <slot/>
    </div>

    <slot v-if="noBody"/>

    <div
      v-if="$slots.footer"
      :class="footerClasses"
      class="card-footer">
      <slot name="footer"/>
    </div>
  </div>
</template>
<script>
export default {
  name: 'Card',
  props: {
    type: {
      type: String,
      description: 'Card type',
      default: null
    },
    gradient: {
      type: String,
      description: 'Card background gradient type (warning,danger etc)',
      default: null
    },
    hover: {
      type: Boolean,
      description: 'Whether card should move on hover',
      default: null
    },
    shadow: {
      type: Boolean,
      description: 'Whether card has shadow',
      default: null
    },
    shadowSize: {
      type: String,
      description: 'Card shadow size',
      default: null
    },
    noBody: {
      type: Boolean,
      default: false,
      description: 'Whether card should have wrapper body class'
    },
    bodyClasses: {
      type: [String, Object, Array],
      description: 'Card body css classes',
      default: null
    },
    headerClasses: {
      type: [String, Object, Array],
      description: 'Card header css classes',
      default: null
    },
    footerClasses: {
      type: [String, Object, Array],
      description: 'Card footer css classes',
      default: null
    }
  }
}
</script>
<style>
</style>
