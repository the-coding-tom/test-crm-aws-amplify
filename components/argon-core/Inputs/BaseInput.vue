<template>
  <div class="form-group">
    <slot name="label">
      <label
        v-if="label"
        :class="labelClasses">
        {{ label }}
      </label>
    </slot>
    <div
      :class="[
        {'input-group': hasIcon},
        {'focused': focused},
        {'input-group-alternative': alternative},
        {'has-label': label || $slots.label},
        inputGroupClasses
    ]">
      <div
        v-if="prependIcon || $slots.prepend"
        class="input-group-prepend">
        <span class="input-group-text">
          <slot name="prepend">
            <i :class="prependIcon"/>
          </slot>
        </span>
      </div>
      <slot v-bind="slotData">
        <input
          :value="value"
          :type="type"
          v-bind="$attrs"
          :valid="!error"
          :required="required"
          :class="[{'is-valid': valid === true}, {'is-invalid': error}, inputClasses]"
          class="form-control"
          v-on="listeners">
      </slot>
      <div
        v-if="appendIcon || $slots.append"
        class="input-group-append">
        <span class="input-group-text">
          <slot name="append">
            <i :class="appendIcon"/>
          </slot>
        </span>
      </div>
      <slot name="infoBlock"/>
      <slot name="error">
        <div
          v-if="error"
          class="invalid-feedback"
          style="display: block;">
          {{ error }}
        </div>
      </slot>
      <slot name="success">
        <div
          v-if="!error && valid"
          class="valid-feedback">
          {{ successMessage }}
        </div>
      </slot>
    </div>
  </div>
</template>
<script>
export default {
  name: 'BaseInput',
  inheritAttrs: false,
  props: {
    required: {
      type: Boolean,
      description: 'Whether input is required (adds an asterix *)',
      default: true
    },
    group: {
      type: Boolean,
      description:
        'Whether input is an input group (manual override in special cases)',
      default: null
    },
    valid: {
      type: Boolean,
      description: 'Whether is valid',
      default: undefined
    },
    alternative: {
      type: Boolean,
      description: 'Whether input is of alternative layout',
      default: false
    },
    label: {
      type: String,
      description: 'Input label (text before input)',
      default: null
    },
    error: {
      type: String,
      description: 'Input error (below input)',
      default: null
    },
    successMessage: {
      type: String,
      description: 'Input success message',
      default: 'Looks good!'
    },
    labelClasses: {
      type: String,
      description: 'Input label css classes',
      default: 'form-control-label'
    },
    inputClasses: {
      type: String,
      description: 'Input css classes',
      default: null
    },
    inputGroupClasses: {
      type: String,
      description: 'Input group css classes',
      default: null
    },
    value: {
      type: [String, Number],
      description: 'Input value',
      default: null
    },
    type: {
      type: String,
      description: 'Input type',
      default: 'text'
    },
    appendIcon: {
      type: String,
      description: 'Append icon (right)',
      default: null
    },
    prependIcon: {
      type: String,
      description: 'Prepend icon (left)',
      default: null
    }
  },
  data() {
    return {
      focused: false
    }
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue,
        focus: this.onFocus,
        blur: this.onBlur
      }
    },
    slotData() {
      return {
        focused: this.focused,
        error: this.error,
        ...this.listeners
      }
    },
    hasIcon() {
      const { append, prepend } = this.$slots
      return (
        append !== undefined ||
        prepend !== undefined ||
        this.appendIcon !== undefined ||
        this.prependIcon !== undefined ||
        this.group
      )
    }
  },
  methods: {
    updateValue(evt) {
      let value = evt.target.value
      this.$emit('input', value)
    },
    onFocus(evt) {
      this.focused = true
      this.$emit('focus', evt)
    },
    onBlur(evt) {
      this.focused = false
      this.$emit('blur', evt)
    }
  }
}
</script>
<style>
</style>
