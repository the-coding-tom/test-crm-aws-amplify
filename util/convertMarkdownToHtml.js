export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
