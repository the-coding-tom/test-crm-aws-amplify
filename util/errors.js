export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
