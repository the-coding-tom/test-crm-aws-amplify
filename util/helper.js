const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
