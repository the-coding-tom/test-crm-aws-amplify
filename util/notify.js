const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
