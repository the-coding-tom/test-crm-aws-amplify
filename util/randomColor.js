const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

export default randomColor
