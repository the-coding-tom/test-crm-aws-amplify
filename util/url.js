export const getQueryParams = url => {
  const urlParam = require('url')

  const parsedUrl = urlParam.parse(url)

  return parsedUrl.search
}
