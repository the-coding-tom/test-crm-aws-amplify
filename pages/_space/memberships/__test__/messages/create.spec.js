import CreateMessagePage from '@/pages/_space/memberships/messages/create.vue'
import ViewMessagePage from '@/pages/_space/memberships/messages/_id/index.vue'
import BaseInput from '~/components/argon-core/Inputs/BaseInput.vue'
import BaseHeader from '~/components/argon-core/BaseHeader'
import BootstrapVue from 'bootstrap-vue'
import BasePagination from '~/components/argon-core/BasePagination'
import Card from '~/components/argon-core/Cards/Card.vue'
import Badge from '~/components/argon-core/Badge.vue'
import DatePicker from 'vue2-datepicker'
import Vue2Filters from 'vue2-filters'
import { Editor } from '@toast-ui/vue-editor'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'
import moment from 'moment'

import currentSpace from '@/pages/_space/__test__/resources/current-space.json'
import allMemberships from '@/pages/_space/memberships/__test__/resources/all-memberships.json'

describe('Page: /:space/memberships/messages/create | Memberships Send Message Page', () => {
  let NuxtStore, store, wrapper, $email, $membership, router

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $email = {
      createEmail: jest.fn().mockResolvedValue({
        data: { id: 'e2e5f8d3-b2c6-4a77-a95c-27d77b8287b0' }
      })
    }

    $membership = {
      getAllMemberships: jest
        .fn()
        .mockResolvedValue({ data: cloneDeep(allMemberships) })
    }

    // mocking 'scrollIntoView' for tui-editor
    window.HTMLElement.prototype.scrollIntoView = jest.fn()
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()
    store.commit('space/setSpace', currentSpace)

    router = new VueRouter({
      routes: [
        {
          name: 'space-memberships-messages-id',
          component: ViewMessagePage,
          path: '_space/memberships/messages/:id'
        }
      ]
    })

    const asyncDataResponse = await CreateMessagePage.asyncData({
      store,
      $email,
      $membership
    })

    wrapper = createWrapper(
      CreateMessagePage,
      store,
      router,
      {
        searching: false,
        emails: [],
        subject: '',
        content: '',
        loading: false,
        ...asyncDataResponse
      },
      {
        stubs: {
          'nuxt-link': true,
          'no-ssr': true,
          'client-only': true,
          transition: false,
          'b-popover': true
        },
        mocks: {
          $email: $email,
          $moment: moment
        }
      },
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.use(Vue2Filters)
        vueInstance.component(BasePagination.name, BasePagination)
        vueInstance.component(BaseHeader.name, BaseHeader)
        vueInstance.component(BaseInput.name, BaseInput)
        vueInstance.component(Card.name, Card)
        vueInstance.component('date-picker', DatePicker)
        vueInstance.component(Badge.name, Badge)
        vueInstance.component('Editor', Editor)
      }
    )
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('sends email', async () => {
    const emailInfo = {
      subject: 'Write Test People !!!',
      content:
        "Writing test should have a different feeling. A ring to it. Unfortunately, it doesn't feel like that. How come people swear they enjoy writing it ???"
    }

    //  select user
    const users = wrapper.vm.$el.querySelectorAll('.el-select-dropdown__item')
    const evt = document.createEvent('MouseEvents')
    evt.initEvent('mouseenter')

    users[0].dispatchEvent(evt)
    users[0].click()
    await flushPromises()

    users[1].dispatchEvent(evt)
    users[1].click()

    await flushPromises()

    // enter email subject
    wrapper.find('#subject').setValue(emailInfo.subject)

    // set email content
    wrapper.setData({
      content: emailInfo.content
    })

    wrapper.find('form').trigger('submit.prevent')
    await flushPromises()

    expect($email.createEmail).toHaveBeenCalled()
    expect($email.createEmail.mock.calls[0][0].subject).toBe(emailInfo.subject)
    expect($email.createEmail.mock.calls[0][0].content).toEqual(
      expect.stringContaining(emailInfo.content)
    )
    expect($email.createEmail.mock.calls[0][0].emails).toContain(
      allMemberships[0].email
    )
    expect($email.createEmail.mock.calls[0][0].emails).toContain(
      allMemberships[1].email
    )
  })
})
