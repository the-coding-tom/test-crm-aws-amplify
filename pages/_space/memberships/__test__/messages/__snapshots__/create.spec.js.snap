// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/memberships/messages/create | Memberships Send Message Page renders properly 1`] = `
<div>
  <form class="">
    <div class="header pb-6">
      <div class="container-fluid">
        <div class="header-body">
          <div class="d-flex justify-content-between align-items-center py-4">
            <h1 class="page-title"><span>Messages</span> <b>â€¢</b> New</h1> <button type="submit" class="btn btn-primary"><i class="fa fa-paper-plane"></i> Send Email</button>
          </div>
        </div>
      </div>
    </div>
    <div class="container-fluid mt--6 mr-b-30">
      <div class="card-deck flex-column flex-xl-row">
        <div class="card">
          <!---->
          <div class="card-body">
            <div class="m-n25 sh-message-box">
              <table class="table">
                <tbody>
                  <tr>
                    <td>
                      <div class="form-group">
                        <!---->
                        <div class="input-group">
                          <!----> <span class="mr-r-10">To</span>
                          <div class="el-select" required="required">
                            <div class="el-select__tags" style="max-width: -32px; width: 100%;">
                              <!---->
                              <transition-group-stub></transition-group-stub><input type="text" autocomplete="off" class="el-select__input" style="flex-grow: 1; width: -0.625%; max-width: -42px;">
                            </div>
                            <div class="el-input el-input--suffix">
                              <!----><input tabindex="-1" type="text" readonly="readonly" autocomplete="off" placeholder="Add Member(s)" class="el-input__inner" style="height: 40px;">
                              <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-"></i><!----><!----><!----><!----><!----></span>
                              <!----></span>
                              <!---->
                              <!---->
                            </div>
                            <div class="el-select-dropdown el-popper is-multiple" style="display: none;">
                              <div class="el-scrollbar" style="">
                                <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
                                  <ul class="el-scrollbar__view el-select-dropdown__list">
                                    <!---->
                                    <li class="el-select-dropdown__item"><span>Ashwin Ravichandran</span></li>
                                    <li class="el-select-dropdown__item"><span>Bubu Nyavor</span></li>
                                  </ul>
                                </div>
                                <div class="el-scrollbar__bar is-horizontal">
                                  <div class="el-scrollbar__thumb" style="transform: translateX(0%); webkit-transform: translateX(0%);"></div>
                                </div>
                                <div class="el-scrollbar__bar is-vertical">
                                  <div class="el-scrollbar__thumb" style="transform: translateY(0%); webkit-transform: translateY(0%);"></div>
                                </div>
                              </div>
                              <!---->
                            </div>
                          </div>
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </div>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <div class="form-inline"><label for="" class="mr-r-10">Subject</label> <input id="subject" type="text" placeholder="Add Subject for Email Here" required="required" class="fg-1 form-control"></div>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <div>
                        <client-only-stub>
                          <div value="" style="box-sizing: border-box; height: 300px;">
                            <div class="tui-editor-defaultUI">
                              <div class="te-toolbar-section">
                                <div class="te-markdown-tab-section" style="display: none;">
                                  <div class="te-tab"><button type="button" data-index="0" class="te-tab-active">Write</button><button type="button" data-index="1">Preview</button></div>
                                </div>
                                <div class="tui-editor-defaultUI-toolbar"><button class="tui-heading tui-toolbar-icons" type="button"></button><button class="tui-bold tui-toolbar-icons" type="button"></button><button class="tui-strike tui-toolbar-icons" type="button"></button>
                                  <div class="tui-toolbar-divider"></div><button class="tui-hrline tui-toolbar-icons" type="button"></button><button class="tui-quote tui-toolbar-icons" type="button"></button>
                                  <div class="tui-toolbar-divider"></div><button class="tui-ul tui-toolbar-icons" type="button"></button><button class="tui-ol tui-toolbar-icons" type="button"></button><button class="tui-task tui-toolbar-icons" type="button"></button><button class="tui-indent tui-toolbar-icons" type="button"></button><button class="tui-outdent tui-toolbar-icons" type="button"></button>
                                  <div class="tui-toolbar-divider"></div><button class="tui-table tui-toolbar-icons" type="button"></button><button class="tui-image tui-toolbar-icons" type="button"></button><button class="tui-link tui-toolbar-icons" type="button"></button>
                                  <div class="tui-toolbar-divider"></div><button class="tui-code tui-toolbar-icons" type="button"></button><button class="tui-codeblock tui-toolbar-icons" type="button"></button>
                                  <div class="tui-toolbar-divider" style="display: none;"></div><button class="tui-scrollsync active" type="button" style="display: none;"></button>
                                  <div class="tui-popup-wrapper te-dropdown-toolbar" style="display: none;">
                                    <div class="tui-popup-body">
                                      <div class="tui-editor-defaultUI-toolbar"></div>
                                    </div>
                                  </div>
                                  <div class="tui-popup-wrapper te-popup-add-table" style="display: none; position: absolute; width: 181px;">
                                    <div class="tui-popup-body">
                                      <div class="te-table-selection">
                                        <div class="te-table-header" style="height: 17px; width: 151px;"></div>
                                        <div class="te-table-body" style="height: 120px; width: 151px;"></div>
                                        <div class="te-selection-area"></div>
                                      </div>
                                      <p class="te-description"></p>
                                    </div>
                                  </div>
                                  <div class="tui-popup-wrapper te-heading-add" style="display: none; position: absolute;">
                                    <div class="tui-popup-body">
                                      <ul>
                                        <li data-value="1" data-type="Heading">
                                          <h1>Heading 1</h1>
                                        </li>
                                        <li data-value="2" data-type="Heading">
                                          <h2>Heading 2</h2>
                                        </li>
                                        <li data-value="3" data-type="Heading">
                                          <h3>Heading 3</h3>
                                        </li>
                                        <li data-value="4" data-type="Heading">
                                          <h4>Heading 4</h4>
                                        </li>
                                        <li data-value="5" data-type="Heading">
                                          <h5>Heading 5</h5>
                                        </li>
                                        <li data-value="6" data-type="Heading">
                                          <h6>Heading 6</h6>
                                        </li>
                                        <li data-type="Paragraph">
                                          <div>Paragraph</div>
                                        </li>
                                      </ul>
                                    </div>
                                  </div>
                                </div>
                              </div>
                              <div class="te-editor-section">
                                <div class="tui-editor te-ww-mode">
                                  <div class="te-md-container te-preview-style-tab">
                                    <div class="te-editor te-tab-active"><textarea style="display: none;"></textarea>
                                      <div class="CodeMirror cm-s-default CodeMirror-wrap" style="min-height: 300px;">
                                        <div style="overflow: hidden; position: relative; width: 3px; height: 0px;"><textarea style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;" autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0"></textarea></div>
                                        <div class="CodeMirror-vscrollbar" tabindex="-1" cm-not-content="true">
                                          <div style="min-width: 1px;"></div>
                                        </div>
                                        <div class="CodeMirror-hscrollbar" tabindex="-1" cm-not-content="true">
                                          <div style="height: 100%; min-height: 1px;"></div>
                                        </div>
                                        <div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div>
                                        <div class="CodeMirror-gutter-filler" cm-not-content="true"></div>
                                        <div class="CodeMirror-scroll" tabindex="-1">
                                          <div class="CodeMirror-sizer" style="margin-left: 0px;">
                                            <div style="position: relative;">
                                              <div class="CodeMirror-lines" role="presentation">
                                                <div style="position: relative; outline: none;" role="presentation">
                                                  <div class="CodeMirror-measure"><pre class="CodeMirror-line-like"><span>xxxxxxxxxx</span></pre>
                                                  </div>
                                                  <div class="CodeMirror-measure"></div>
                                                  <div style="position: relative; z-index: 1;"></div>
                                                  <div class="CodeMirror-cursors"></div>
                                                  <div class="CodeMirror-code" role="presentation"></div>
                                                </div>
                                              </div>
                                            </div>
                                          </div>
                                          <div style="position: absolute; height: 50px; width: 1px;"></div>
                                          <div class="CodeMirror-gutters" style="display: none;"></div>
                                        </div>
                                      </div>
                                    </div>
                                    <div class="te-md-splitter"></div>
                                    <div class="te-preview" style="min-height: 300px;">
                                      <div class="tui-editor-contents"></div>
                                    </div>
                                  </div>
                                  <div class="te-ww-container">
                                    <div class="te-editor" style="position: relative;">
                                      <div contenteditable="true" class="tui-editor-contents tui-editor-contents-placeholder" style="min-height: 300px;"><br>
                                        <div><br></div>
                                      </div>
                                      <div class="te-ww-block-overlay code-block-header" style="position: absolute; display: none; z-index: 1;"><span>text</span><button type="button">Editor</button></div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                              <div class="te-mode-switch-section" style="display: none;">
                                <div class="te-mode-switch"><button class="te-switch-button markdown" type="button">Markdown</button><button class="te-switch-button wysiwyg active" type="button">WYSIWYG</button></div>
                              </div>
                              <div class="tui-popup-wrapper te-popup-add-link tui-editor-popup" style="display: none;">
                                <div class="tui-popup-header">
                                  <span class="tui-popup-title">Insert link</span>
                                  <div class="tui-popup-header-buttons">
                                    <button type="button" class="tui-popup-close-button"></button>
                                  </div>
                                </div>
                                <div class="tui-popup-body">
                                  <label for="url">URL</label>
                                  <input type="text" class="te-url-input">
                                  <label for="linkText">Link text</label>
                                  <input type="text" class="te-link-text-input">
                                  <div class="te-button-section">
                                    <button type="button" class="te-ok-button">OK</button>
                                    <button type="button" class="te-close-button">Cancel</button>
                                  </div>
                                </div>
                              </div>
                              <div class="tui-popup-wrapper te-popup-add-image tui-editor-popup" style="display: none;">
                                <div class="tui-popup-header">
                                  <span class="tui-popup-title">Insert image</span>
                                  <div class="tui-popup-header-buttons">
                                    <button type="button" class="tui-popup-close-button"></button>
                                  </div>
                                </div>
                                <div class="tui-popup-body">
                                  <div class="te-tab-section">
                                    <div class="te-tab"><button type="button" data-index="0" class="te-tab-active">File</button><button type="button" data-index="1">URL</button></div>
                                  </div>
                                  <div class="te-url-type">
                                    <label for="">Image URL</label>
                                    <input type="text" class="te-image-url-input">
                                  </div>
                                  <div class="te-file-type te-tab-active">
                                    <label for="">Select image file</label>
                                    <input type="file" class="te-image-file-input" accept="image/*">
                                  </div>
                                  <label for="url">Description</label>
                                  <input type="text" class="te-alt-text-input">
                                  <div class="te-button-section">
                                    <button type="button" class="te-ok-button">OK</button>
                                    <button type="button" class="te-close-button">Cancel</button>
                                  </div>
                                </div>
                              </div>
                              <div class="tui-popup-wrapper te-popup-table-utils" style="display: none;">
                                <div class="tui-popup-body">
                                  <button type="button" class="te-table-add-row">Add row</button>
                                  <button type="button" class="te-table-add-col">Add col</button>
                                  <button type="button" class="te-table-remove-row">Remove row</button>
                                  <button type="button" class="te-table-remove-col">Remove col</button>
                                  <hr>
                                  <button type="button" class="te-table-col-align-left">Align left</button>
                                  <button type="button" class="te-table-col-align-center">Align center</button>
                                  <button type="button" class="te-table-col-align-right">Align right</button>
                                  <hr>
                                  <button type="button" class="te-table-remove">Remove table</button>
                                </div>
                              </div>
                              <div class="tui-popup-wrapper te-popup-code-block-languages" style="display: none; z-index: 10000;">
                                <div class="tui-popup-body"></div>
                              </div>
                              <div class="tui-popup-modal-background tui-popup-code-block-editor" style="display: none;">
                                <div class="tui-popup-wrapper">
                                  <div class="tui-popup-header">
                                    <span class="tui-popup-title">CodeBlock Editor<span class="te-input-language" style="display: none;"><input type="text" maxlength="20" placeholder="Choose language"></span></span>
                                    <div class="tui-popup-header-buttons">
                                      <button type="button" class="popup-editor-toggle-scroll active" style="display: inline-block;"></button>
                                      <button type="button" class="popup-editor-toggle-preview active"></button>
                                      <button type="button" class="popup-editor-toggle-fit"></button>
                                      <button type="button" class="tui-popup-close-button"></button>
                                    </div>
                                  </div>
                                  <div class="tui-popup-body">
                                    <div class="popup-editor-body">
                                      <div class="tui-split-scroll scroll-sync">
                                        <div class="tui-split-scroll-wrapper">
                                          <div class="tui-split-scroll-content">
                                            <div class="popup-editor-editor-wrapper tui-split-content-left"><textarea style="display: none;"></textarea>
                                              <div class="CodeMirror cm-s-default CodeMirror-wrap">
                                                <div style="overflow: hidden; position: relative; width: 3px; height: 0px;"><textarea style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;" autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0"></textarea></div>
                                                <div class="CodeMirror-vscrollbar" tabindex="-1" cm-not-content="true">
                                                  <div style="min-width: 1px;"></div>
                                                </div>
                                                <div class="CodeMirror-hscrollbar" tabindex="-1" cm-not-content="true">
                                                  <div style="height: 100%; min-height: 1px;"></div>
                                                </div>
                                                <div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div>
                                                <div class="CodeMirror-gutter-filler" cm-not-content="true"></div>
                                                <div class="CodeMirror-scroll" tabindex="-1">
                                                  <div class="CodeMirror-sizer" style="margin-left: 0px;">
                                                    <div style="position: relative;">
                                                      <div class="CodeMirror-lines" role="presentation">
                                                        <div style="position: relative; outline: none;" role="presentation">
                                                          <div class="CodeMirror-measure"><pre class="CodeMirror-line-like"><span>xxxxxxxxxx</span></pre>
                                                          </div>
                                                          <div class="CodeMirror-measure"></div>
                                                          <div style="position: relative; z-index: 1;"></div>
                                                          <div class="CodeMirror-cursors"></div>
                                                          <div class="CodeMirror-code" role="presentation"></div>
                                                        </div>
                                                      </div>
                                                    </div>
                                                  </div>
                                                  <div style="position: absolute; height: 50px; width: 1px;"></div>
                                                  <div class="CodeMirror-gutters" style="display: none;"></div>
                                                </div>
                                              </div>
                                            </div>
                                            <div class="tui-split-content-right" style="top: 0px;">
                                              <div class="tui-editor-contents"></div>
                                            </div>
                                          </div>
                                          <div class="tui-splitter"></div>
                                        </div>
                                      </div>
                                    </div>
                                    <div class="te-button-section">
                                      <button type="button" class="te-ok-button">OK</button>
                                      <button type="button" class="te-close-button">Cancel</button>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </client-only-stub>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div> <a href="#" target="_self" class=""><i class="fa fa-angle-left mr-r-10"></i> Back</a>
    </div>
  </form>
</div>
`;
